File: app/src/main/java/com/mnj/java_solutions/LargestOddNumberInStringLC1903.java
package com.mnj.java_solutions;

public class LargestOddNumberInStringLC1903 {

    public void largestOddNumber() {
        String num = "78389";

        for (int i = num.length() - 1; i >= 0; i--) {
            char c = num.charAt(i);
            if (c % 2 == 1) {
                System.out.println("==>> largest odd number is :" + num.substring(0, i + 1));
                return;
            }
        }
        System.out.println("==>> No odd number exists...");
    }
}

File: app/src/main/java/com/mnj/java_solutions/LargestOddNumberInStringLC1903.kt
package com.mnj.java_solutions

class LargestOddNumberInStringLC1903 {
    fun largestOddNumber() {
        val num = "78389"
        for (i in num.length - 1 downTo 0) {
            val c = num[i]
            if (c.code % 2 == 1) {
                println("==>> largest odd number is :" + num.substring(0, i + 1))
                return
            }
        }
        println("==>> No odd number exists...")
    }
}